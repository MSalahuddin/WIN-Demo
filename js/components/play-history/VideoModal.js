import React from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import { SafeAreaView } from 'react-native';
import PropTypes from 'prop-types';
import Video from 'react-native-video';

import IconButton from '../common/IconButton';
import { scale } from '../../platformUtils';
import { closeBlack } from '../../../assets/images';
import { color } from '../../styles';
import { logEvent } from '../../amplitudeUtils';
import { ANALYTICS_EVENTS, ANALYTICS_PROPERTIES } from '../../constants';

const StyledModal = styled(Modal)`
  background-color: ${color.black};
  margin: 0;
  justify-content: flex-start;
`;

const Header = styled.View`
  flex-direction: row;
  padding-horizontal: ${scale(24)};
  position: absolute;
  width: 100%;
  justify-content: flex-end;
  z-index: 1;
`;

const VideoWrapper = styled(Video)`
  top: 25%;
  width: 100%;
  height: 50%;
  background-color: ${color.black};
`;

const VideoModal = ({ isVisible, setVisible, videoInfo, videoOnError, ...rest }) => {
  const { gameRoundId, prizeName, replayUrl } = videoInfo;
  const trackWatchGamePlay = () => {
    logEvent(ANALYTICS_EVENTS.WATCHED_GAME_PLAY, {
      [ANALYTICS_PROPERTIES.PRIZE_NAME]: prizeName,
      [ANALYTICS_PROPERTIES.GAME_ROUND_ID]: gameRoundId
    });
  };
  return (
    <SafeAreaView>
      <StyledModal isVisible={isVisible} {...rest}>
        <Header>
          <IconButton testID="cancel-button" onPress={() => setVisible(false)} icon={closeBlack} size={32} />
        </Header>
        <VideoWrapper
          controls
          fullscreen
          onLoad={trackWatchGamePlay}
          onError={videoOnError}
          source={{ uri: replayUrl }}
          resizeMode="cover"
        />
      </StyledModal>
    </SafeAreaView>
  );
};

VideoModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  setVisible: PropTypes.func.isRequired,
  videoOnError: PropTypes.func.isRequired,
  videoInfo: PropTypes.shape({
    replayUrl: PropTypes.string.isRequired,
    gameRoundId: PropTypes.number.isRequired,
    prizeName: PropTypes.string.isRequired
  }).isRequired
};

export default VideoModal;
