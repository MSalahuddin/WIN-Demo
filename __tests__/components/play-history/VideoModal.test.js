import React from 'react';
import { render } from 'react-native-testing-library';

import VideoModal from '../../../js/components/play-history/VideoModal';

describe('Component VideoModal ', () => {
  const setVisible = jest.fn();
  const videoOnError = jest.fn();
  const videoInfo = {
    gameRoundId: 0,
    prizeName: '',
    replayUrl: ''
  };

  test('renders the video popup correctly', () => {
    const output = render(
      <VideoModal isVisible setVisible={setVisible} videoInfo={videoInfo} videoOnError={videoOnError} />
    );
    expect(output).toMatchSnapshot();
  });
});
