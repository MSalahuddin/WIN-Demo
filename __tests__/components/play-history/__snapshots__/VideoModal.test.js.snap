// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component VideoModal  renders the video popup correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "backgroundColor": "#000000",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "paddingHorizontal": 36,
              "position": "absolute",
              "width": "100%",
              "zIndex": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 24,
              "marginTop": 48,
              "opacity": 1,
            }
          }
          testID="cancel-button"
        >
          <Image
            opacity={1}
            resizeMode="contain"
            size={32}
            source={1}
            style={
              Array [
                Object {
                  "height": 48,
                  "width": 48,
                },
              ]
            }
          />
        </View>
      </View>
      <Video
        controls={true}
        fullscreen={true}
        onError={[MockFunction]}
        onLoad={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "#000000",
              "height": "50%",
              "top": "25%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
  </Modal>
</RCTSafeAreaView>
`;
